avocado/plugins/runner_nrunner.py

```diff
@@ -160,66 +164,88 @@ def _get_all_runtime_tasks(self, test_suite):
+                job.result_events_dispatcher.map_method('end_test',
+                                                        job.result,
+                                                        test_state)
+
     def run_suite(self, job, test_suite):
         summary = set()
-        if job.timeout > 0:
-            deadline = time.time() + job.timeout
-        else:
-            deadline = None
 
         test_suite.tests, _ = nrunner.check_tasks_requirements(test_suite.tests)
         job.result.tests_total = test_suite.size  # no support for variants yet
-        result_dispatcher = job.result_events_dispatcher
-        status_repo = StatusRepo()
-        for runtime_task in self._get_all_runtime_tasks(test_suite):
-            if deadline is not None and time.time() > deadline:
-                break
-            task = runtime_task.task
-
-            early_state = {
-                'name': task.identifier,
-                'job_logdir': job.logdir,
-                'job_unique_id': job.unique_id,
-            }
-            job.result.start_test(early_state)
-            job.result_events_dispatcher.map_method('start_test',
-                                                    job.result,
-                                                    early_state)
-
-            task.status_services = []
-            for status in task.run():
-                status_repo.process_message(status)
-                result_dispatcher.map_method('test_progress', False)
-
-                if status['status'] not in ["started", "running"]:
-                    break
-
-            # test execution time is currently missing
-            # since 358e800e81 all runners all produce the result in a key called
-            # 'result', instead of 'status'.  But the Avocado result plugins rely
-            # on the current runner approach
-            this_task_data = status_repo.get_task_data(task.identifier)
-            test_state = {'status': this_task_data[-1]['result'].upper()}
-            test_state.update(early_state)
-
-            time_start = this_task_data[0]['time']
-            time_end = this_task_data[-1]['time']
-            time_elapsed = time_end - time_start
-            test_state['time_start'] = time_start
-            test_state['time_end'] = time_end
-            test_state['time_elapsed'] = time_elapsed
-
-            # fake log dir, needed by some result plugins such as HTML
-            test_state['logdir'] = ''
-
-            # Populate task dir
-            base_path = os.path.join(job.logdir, 'test-results')
-            self._populate_task_logdir(base_path,
-                                       task,
-                                       this_task_data,
-                                       job.config.get('core.debug'))
-
-            job.result.check_test(test_state)
-            result_dispatcher.map_method('end_test', job.result, test_state)
+
+        status_server_uri = test_suite.config.get('nrunner.status_server_uri')
```

Comment:
Small suggestion / Up to you: Can we let everything related to status_server inside `nrunner.status_server` section?